src/main/java/gregtech/common/terminal/app/worldprospector/WorldProspectorARApp.java

```diff
@@ -95,6 +143,20 @@ public boolean mouseClicked(int mouseX, int mouseY, int button) {
                 public void handleClientAction(int id, PacketBuffer buffer) {
                     if (id == -1) {
                         selectReference(index, buttonWidget);
+                    }
+                    else if (id == 0) {
+                        updateBlockSelectionAndColor(ItemStack.EMPTY, index, buttonWidget);
+                    }
+                    else if (id == 1) {
+                        PacketBuffer tmpBuffer = new PacketBuffer(buffer.copy());
```

Comment:
Why does the PacketBuffer need to be copied here?

Reply:
`PacketBuffer#read` methods increase a read index which is how you can readInt, readItemStack, readBoolean back to back. I didnt see a way to reset it with a quick glance.

Turns out there is a way....in the clear method.
![image](https://user-images.githubusercontent.com/18713839/227691220-fbc36cbf-50fa-40fc-a9e0-a01a0c4cda94.png)

Reply:
Changed to marking the current reader index, then just resetting the read index